spec/rubocop/cop/rails/helper_instance_variable_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::HelperInstanceVariable do
+  subject(:cop) { described_class.new }
+
+  it 'reports uses of instance variables' do
+    expect_offense(<<-'RUBY'.strip_indent)
+      def welcome_message
+        "Hello #{@user.name}"
+                 ^^^^^ Do not use instance variables in helpers.
+      end
+    RUBY
+  end
+
+  it 'reports instance variable assignments' do
+    expect_offense(<<-RUBY.strip_indent)
+      def welcome_message(user)
+        @user_name = user.name
+        ^^^^^^^^^^^^^^^^^^^^^^ Do not use instance variables in helpers.
```

Comment:
Minor nit: would it be more consistent, and slightly more clear, if the highlight was limited to just the instance variable being assigned?

Reply:
Sure! 🙂

First, use the `ruby-parse` utility provided with the parser gem, to find the locations available for this source map:

```
$ ruby-parse -L -e '@user_name = user.name'
```

You should see something like this:

```
@user_name = user.name
~~~~~~~~~~ name      
           ~ operator            
~~~~~~~~~~~~~~~~~~~~~~ expression
```

So in this case we want to target the `name` location. The `#add_offense` method takes an optional `location` argument that is automatically turned into a range. (Defaults to `:expression`.)

So to add an offense on the relevant part:

```ruby
add_offense(node, location: :name)
```

Reply:
Thanks for that. Fixed in 6dd2d00.